precompiles/crowdloan-rewards/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+				return Err(ExitError::OutOfGas);
+			}
+		}
+
+		let account: Runtime::AccountId = contributor.into();
+		// fetch data from pallet
+		let is_contributor =
+			pallet_crowdloan_rewards::Pallet::<Runtime>::accounts_payable(account).is_some();
+
+		log::trace!(target: "crowldoan-rewards-precompile", "Result from pallet is {:?}", is_contributor);
+
+		let gas_consumed = <Runtime as pallet_evm::Config>::GasWeightMapping::weight_to_gas(
+			<Runtime as frame_system::Config>::DbWeight::get().read,
+		);
+
+		Ok(PrecompileOutput {
+			exit_status: ExitSucceed::Returned,
+			cost: gas_consumed,
+			output: bool_to_solidity_bytes(is_contributor),
+			logs: Default::default(),
+		})
+	}
+
+	fn reward_info(input: &[u8], target_gas: Option<u64>) -> Result<PrecompileOutput, ExitError> {
+		// parse the address
+		let contributor = H160::from_slice(&input[12..32]);
+
+		log::trace!(
+			target: "crowdloan-rewards-precompile",
+			"Checking reward info for {:?}",
+			contributor
+		);
+
+		let account: Runtime::AccountId = contributor.into();
+
+		let gas_consumed = <Runtime as pallet_evm::Config>::GasWeightMapping::weight_to_gas(
+			<Runtime as frame_system::Config>::DbWeight::get().read,
+		);
+
+		// Make sure enough gas
+		if let Some(gas_limit) = target_gas {
+			if gas_consumed > gas_limit {
+				return Err(ExitError::OutOfGas);
+			}
+		}
+
+		// fetch data from pallet
+		let reward_info = pallet_crowdloan_rewards::Pallet::<Runtime>::accounts_payable(account);
+
+		let (total, claimed): (U256, U256) = if let Some(reward_info) = reward_info {
+			let total_reward: u128 = reward_info.total_reward.try_into().map_err(|_| {
+				ExitError::Other("Amount is too large for provided balance type".into())
+			})?;
+			let claimed_reward: u128 = reward_info.claimed_reward.try_into().map_err(|_| {
+				ExitError::Other("Amount is too large for provided balance type".into())
+			})?;
+
+			(total_reward.into(), claimed_reward.into())
+		} else {
+			(0u128.into(), 0u128.into())
+		};
+
+		log::trace!(
+			target: "crowldoan-rewards-precompile", "Result from pallet is {:?}  {:?}",
+			total, claimed
+		);
+
+		let mut buffer = [0u8; 64];
+		buffer[0..32].clone_from_slice(u256_to_solidity_bytes(total).as_slice());
+		buffer[32..64].clone_from_slice(u256_to_solidity_bytes(claimed).as_slice());
+
+		Ok(PrecompileOutput {
+			exit_status: ExitSucceed::Returned,
+			cost: gas_consumed,
+			output: buffer.to_vec(),
+			logs: Default::default(),
+		})
+	}
+
+	fn claim() -> Result<pallet_crowdloan_rewards::Call<Runtime>, ExitError> {
+		Ok(pallet_crowdloan_rewards::Call::<Runtime>::claim())
+	}
+
+	fn update_reward_address(
+		input: &[u8],
+	) -> Result<pallet_crowdloan_rewards::Call<Runtime>, ExitError> {
+		log::trace!(
+			target: "crowdloan-rewards-precompile",
+			"In update_reward_address dispatchable wrapper"
+		);
+		log::trace!(target: "crowdloan-rewards-precompile", "input is {:?}", input);
+		let new_address = parse_account(&input[..32])?;
```

Comment:
Do you need a bounds check here?

Reply:
I can add them yes

Reply:
`parse_account` already checks the input length and returns an error if wrong